allennlp/modules/span_extractors/bidirectional_endpoint_span_extractor.py

```diff
@@ -0,0 +1,230 @@
+    (start_index - end_index), as in the backward direction, the relative positions of the
+    start and end indices of the spans are switched (e.g if you trace the direction that the RNN
+    is going in, you encounter the end of the span first, before the start of the span).
+    All references to ``start`` and ``end`` span indices in this class are with respect
+    to the forward direction.
+
+    Additionally, the width of the spans can be embedded and concatenated on to the
+    final combination.
+
+    The following other types of representation are supported for both the forward and backward
+    directions, assuming that ``x = span_start_embeddings`` and ``y = span_end_embeddings``.
+
+    ``x``, ``y``, ``x*y``, ``x+y``, ``x-y``, ``x/y``, where each of those binary operations
+    is performed elementwise.  You can list as many combinations as you want, comma separated.
+    For example, you might give ``x,y,x*y`` as the ``combination`` parameter to this class.
+    The computed similarity function would then be ``[x; y; x*y]``, which can then be optionally
+    concatenated with an embedded representation of the width of the span.
+
+    Parameters
+    ----------
+    input_dim : ``int``, required.
+        The final dimension of the ``sequence_tensor``.
+    forward_combination : str, optional (default = "y-x").
+        The method used to combine the ``forward_start_embeddings`` and ``forward_end_embeddings``
+        for the forward direction of the bidirectional representation.
+        See above for a full description.
+    backward_combination : str, optional (default = "x-y").
+        The method used to combine the ``backward_start_embeddings`` and ``backward_end_embeddings``
+        for the backward direction of the bidirectional representation.
+        See above for a full description.
+    num_width_embeddings : ``int``, optional (default = None).
+        Specifies the number of buckets to use when representing
+        span width features.
+    span_width_embedding_dim : ``int``, optional (default = None).
+        The embedding size for the span_width features.
+    bucket_widths : ``bool``, optional (default = False).
+        Whether to bucket the span widths into log-space buckets. If ``False``,
+        the raw span widths are used.
+    use_sentinels : ``bool``, optional (default = ``True``).
+        If ``True``, sentinels are used to represent exclusive span indices for the elements
+        in the first and last positions in the sequence (as the exclusive indices for these
+        elements are outside of the the sequence boundary). This is not strictly necessary,
+        as you may know that your exclusive start and end indices are always within your sequence
+        representation, such as if you have appended/prepended <START> and <END> tokens to your
+        sequence (equivelent to learning a sentinel representation in this class, but more accessable).
```

Comment:
This isn't totally true, because the LSTM states will be different if you've got start and end tokens.  I'd probably just take out the parenthetical here.